File: source/phpfiles/weha/diff/Insertion.php
<?php
include_once('BasicEdit.php');

class Insertion extends BasicEdit {

	public function __construct($pos, array $content) {
		$this->oldPos = -1;
		$this->newPos = $pos;
		$this->oldTokens = null;
		$this->newTokens = $content;
	}
	
	public function setStartPos($pos) {
		$this->newPos = $pos;
	}
	
	public function getPos() {
		return $this->newPos;
	}
	
	public function getContent() {
		return $this->newTokens;
	}

	public function setContent($actionContent) {
		$this->newTokens = $actionContent;		
	}
	
	public function getLength() {
		return count($this->newTokens);
	}
	
	public function getDescription() {
		return sprintf("Ins(%s, %d)", implode('', $this->newTokens), $this->newPos);
	}

	public function getNewPos() {
		return $this->newPos;
	}

	public function getOldPos() {
		return $this->oldPos;
	}
}

File: source/weha/wikianalysis/diff/token/Insertion.java
package mo.umac.wikianalysis.diff.token;

import mo.umac.wikianalysis.lexer.WikiToken;

public class Insertion extends BasicEdit {

	public Insertion(int pos, WikiToken[] content) {
		super();
		this.oldPos = -1;
		this.newPos = pos;
		this.oldTokens = null;
		this.newTokens = content;
	}
	
	public void setStartPos(int pos) {
		this.newPos = pos;
	}
	
	public int getPos() {
		return newPos;
	}
	
	public WikiToken[] getContent() {
		return newTokens;
	}

	public void setContent(WikiToken[] actionContent) {
		this.newTokens = actionContent;		
	}
	
	public int getLength() {
		return newTokens.length;
	}
	
	public String getDescription() {
		return String.format("Ins(%d, %d)", newTokens.length, newPos);
	}

	@Override
	public int getNewPos() {
		return newPos;
	}

	@Override
	public int getOldPos() {
		return oldPos;
	}

	@Override
	public String getLinkedDesc() {
		return String.format("Ins(%d, <a href='#n%d'>%d</a>)", newTokens.length, newPos, newPos);
	}
}
